### PR TITLE
Recreate graal RetryPolicy issue

### LINKED ISSUE #6127: GraalVM - Improve compatibility with native images
**Describe the bug**
Unable to build a Spring Boot 3.2 application using Java 21 and GraalVM 22.3, using AOT native-image. 

**Steps to reproduce**

```
<dependency>
    <groupId>io.opentelemetry</groupId>
    <artifactId>opentelemetry-exporter-otlp</artifactId>
</dependency>
```

```java 
@Configuration
public class ObservationConfiguration {

    @Bean
    public SpanExporter otlpSpanExporter() {
        return OtlpGrpcSpanExporter.builder()
            .setEndpoint("http://localhost:4317")
            .build();
    }

}
```

**What did you expect to see?**
My understanding is that opentelemetry has native-image support. So I'd expect the application to build without crashing or throwing an exception. 

**What did you see instead?**
```
com.oracle.svm.core.util.VMError$HostedError: com.oracle.svm.core.util.UserError$UserException: Image heap writing found a class not seen during static analysis. Did a static field or an object referenced from a static field change during native image generation? For example, a lazily initialized cache could have been initialized during image generation, in which case you need to force eager initialization of the cache before static analysis or reset the cache using a field value recomputation.
    class: io.opentelemetry.sdk.common.export.RetryPolicy
  reachable through:
    object: [Ljava.lang.Class;@73d05767  of class: java.lang.Class[]
    object: com.oracle.svm.core.code.ImageCodeInfo@3b2d610e  of class: com.oracle.svm.core.code.ImageCodeInfo
    root: com.oracle.svm.core.code.ImageCodeInfo.prepareCodeInfo()
```

**What version and what artifacts are you using?**
```
 +- io.micrometer:micrometer-tracing-bridge-otel:jar:1.2.0:compile
[INFO] |  +- io.micrometer:micrometer-tracing:jar:1.2.0:compile
[INFO] |  |  +- io.micrometer:context-propagation:jar:1.1.0:compile
[INFO] |  |  \- aopalliance:aopalliance:jar:1.0:compile
[INFO] |  +- io.opentelemetry:opentelemetry-api:jar:1.31.0:compile
[INFO] |  |  \- io.opentelemetry:opentelemetry-context:jar:1.31.0:compile
[INFO] |  +- io.opentelemetry.semconv:opentelemetry-semconv:jar:1.21.0-alpha:compile
[INFO] |  +- io.opentelemetry:opentelemetry-sdk-common:jar:1.31.0:compile
[INFO] |  +- io.opentelemetry:opentelemetry-sdk-trace:jar:1.31.0:compile
[INFO] |  +- io.opentelemetry:opentelemetry-sdk:jar:1.31.0:compile
[INFO] |  +- io.opentelemetry.instrumentation:opentelemetry-instrumentation-api-semconv:jar:1.31.0-alpha:compile
[INFO] |  |  +- io.opentelemetry.instrumentation:opentelemetry-instrumentation-api:jar:1.31.0:compile
[INFO] |  |  \- io.opentelemetry:opentelemetry-extension-incubator:jar:1.31.0-alpha:runtime
[INFO] |  \- io.opentelemetry:opentelemetry-extension-trace-propagators:jar:1.31.0:compile
[INFO] +- io.micrometer:micrometer-registry-otlp:jar:1.12.0:compile
[INFO] |  +- io.micrometer:micrometer-core:jar:1.12.0:compile
[INFO] |  |  +- org.hdrhistogram:HdrHistogram:jar:2.1.12:runtime
[INFO] |  |  \- org.latencyutils:LatencyUtils:jar:2.0.3:runtime
[INFO] |  \- io.opentelemetry.proto:opentelemetry-proto:jar:1.0.0-alpha:runtime
[INFO] |     \- com.google.protobuf:protobuf-java:jar:3.23.4:runtime
[INFO] +- io.opentelemetry:opentelemetry-exporter-otlp:jar:1.31.0:compile
[INFO] |  +- io.opentelemetry:opentelemetry-sdk-metrics:jar:1.31.0:compile
[INFO] |  +- io.opentelemetry:opentelemetry-sdk-logs:jar:1.31.0:compile
[INFO] |  |  \- io.opentelemetry:opentelemetry-api-events:jar:1.31.0-alpha:runtime
[INFO] |  +- io.opentelemetry:opentelemetry-exporter-otlp-common:jar:1.31.0:runtime
[INFO] |  |  \- io.opentelemetry:opentelemetry-exporter-common:jar:1.31.0:runtime
[INFO] |  +- io.opentelemetry:opentelemetry-exporter-sender-okhttp:jar:1.31.0:runtime
```

**Environment**
Compiler: "Graalvm 22.3"
MacOS

**Additional context**
I fixed the issue locally by adding;

```
{
  "name":"io.opentelemetry.sdk.common.export.AutoValue_RetryPolicy",
  "queryAllDeclaredMethods":true
},
{
  "name":"io.opentelemetry.sdk.common.export.RetryPolicy",
  "queryAllDeclaredMethods":true
}
```

### PR DESCRIPTION
**Pull request by jack-berg**
Resolves #6127.

### CODE REVIEW

**Comment by trask** [comment]
is this a workaround for a Graal bug? I don't see any reflective access for `RetryPolicy`

cc @jeanbisutti

**Comment by jack-berg** [comment]
> is this a workaround for a Graal bug? I don't see any reflective access for RetryPolicy

That was my conclusion in this [comment](https://github.com/open-telemetry/opentelemetry-java/issues/6127#issuecomment-1883592741). The fact that this doesn't occur in a different version of graal does suggest some sort of regression in the compiler.

File: exporters/common/src/main/resources/META-INF/native-image/io.opentelemetry.opentelemetry-exporter-common/reflect-config.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[
+  {
+    "name":"io.opentelemetry.sdk.common.export.AutoValue_RetryPolicy",
```
Comment (trask) [inline exporters/common/src/main/resources/META-INF/native-image/io.opentelemetry.opentelemetry-exporter-common/reflect-config.json:3]: maybe add comment here about why we're adding it even though there's no reflective usage?

**Review by trask — APPROVED** [review]

File: exporters/common/src/main/resources/META-INF/native-image/io.opentelemetry.opentelemetry-exporter-common/reflect-config.json (hunk 1 above)
Comment (jack-berg) [reply exporters/common/src/main/resources/META-INF/native-image/io.opentelemetry.opentelemetry-exporter-common/reflect-config.json:3]: JSON doesn't typically support comments but I can give it a try

Comment (trask) [reply exporters/common/src/main/resources/META-INF/native-image/io.opentelemetry.opentelemetry-exporter-common/reflect-config.json:3]: ugh ya

Comment (trask) [reply exporters/common/src/main/resources/META-INF/native-image/io.opentelemetry.opentelemetry-exporter-common/reflect-config.json:3]: no worries if it doesn't work

Comment (jack-berg) [reply exporters/common/src/main/resources/META-INF/native-image/io.opentelemetry.opentelemetry-exporter-common/reflect-config.json:3]:
```
Error: Error parsing reflection configuration in jar:file:/Users/jberg/code/open-telemetry/opentelemetry-java/exporters/common/build/libs/opentelemetry-exporter-common-1.35.0-SNAPSHOT.jar!/META-INF/native-image/io.opentelemetry.opentelemetry-exporter-common/reflect-config.json:
line 3 column 4 Expected , or } but found /
```
No dice

**Pull request merged**